grc/gui_qt/components/variable_editor.py

```diff
@@ -96,6 +97,18 @@ def contextMenuEvent(self, event):
     def keyPressEvent(self, event):
         super(VariableEditor, self).keyPressEvent(event)
 
+    def handle_click(self, item, col):
+        # we only care about the add/remove icons being clicked
+        if (col != 2):
+            return
+
+        if item.text(0) == "Imports":
+            self.handle_action(VariableEditorAction.ADD_IMPORT)
+        elif item.text(0) == "Variables":
+            self.handle_action(VariableEditorAction.ADD_VARIABLE)
+        else:
+            self.handle_action(VariableEditorAction.DELETE_BLOCK)
```

Comment:
Gut feeling:

Not a fan of using the text displayed on a menu item as "magic value" somewhere else in the code to track which menu item was clicked. That keeps the program untranslatable, and means someone changing appearance unintentionally changes behaviour.

Can we instead have different `handle_click`-like functions, each connected to the `variables` or `imports` object from `__init__`? Don't know whether that's technically possible. If it isn't, one would store `variables` and `imports` in `self`, and compare against the `item`, rather than `item.text(0)`.

Reply:
You're of course right; the items here are just entries in the data model that the tree view displays. My mistake.

I can see the problem with the dangling references. What about we give the items data, i.e., use setData to identify them?

Reply:
I hadn't thought of that, but it is definitely the way to go about it. Im stuck on what the data should be but since we're talking about literally two objects, could we get away with it just being strings? 
i.e.
`imports.setData(0, QtGui.UserRole, "imports")`
since strings are immutable in python i think this will be pretty inexpensive.
Otherwise we can assign some constant value to two fields of the class and set that to be the data. 
i.e.
`__init__():`
`...`
`self.import_const = some_value`
`...`
and
`imports.setData(0, QtGui.UserRole, self.import_const)`

But I'm a fan of the strings because we can be pretty explicit and use something like "Imports Tab", making what is going on very clear.

Reply:
for something that only happens when a user clicks, you really don't have to think that much about efficiency, by the way :) That becomes important when you need to do something a couple thousand times a second.

I do like the constant idea! But for a different reason: You already have the right constants for that: in the `VariableEditorAction` enum

Reply:
Oh wow that's very clever, I'm going to implement the change really quick and then squash and push the changes